lib/stimpack/pack.rb

```diff
@@ -1,53 +1,55 @@
 # frozen_string_literal: true
 
+require "pathname"
+
 module Stimpack
   class Pack
     PACKAGE_FILE = "package.yml"
+    ROOT_PATH = Pathname.new(".")
 
     autoload :Configuration, "stimpack/pack/configuration"
 
-    attr_reader :name
     attr_reader :path
-    attr_reader :engine
+    attr_reader :name
+    attr_accessor :engine
 
-    def self.create(path)
-      new(path) if path.join(PACKAGE_FILE).exist?
-    end
+    def initialize(path, level:, packs_path: nil)
+      if Stimpack.config.max_levels && level > Stimpack.config.max_levels
```

Comment:
Could/should we just hardcode and introduce this configurability when/if we want/need it?

Reply:
This is no longer relevant